apps/src/util/curriculumRecommender/curriculumRecommender.js

```diff
@@ -207,9 +301,9 @@ const hasImportantButNotDesiredTopic = (curriculum, csTopics) => {
   const curriculumTopics = curriculum.cs_topic?.split(',');
   const desiredTopics = csTopics?.split(',');
 
-  if (curriculumTopics && desiredTopics) {
+  if (curriculumTopics) {
     for (const topic of curriculumTopics) {
-      if (IMPORTANT_TOPICS.includes(topic) && !desiredTopics.includes(topic)) {
+      if (IMPORTANT_TOPICS.includes(topic) && !desiredTopics?.includes(topic)) {
```

Comment:
Fixes a scoring bug to ensure that curricula still receive points for having an important topic even if there are no desired topics passed in